### PR TITLE
Add few IOperation unit tests for foreach statement

### LINKED ISSUE #22030: Add more tests for C# and VB foreach loop statements
See https://github.com/dotnet/roslyn/pull/21791#discussion_r137703488.

PR review comment mentions: `Are there tests that use convert collection to generic IEnumerable, or use For Each pattern? Is for each over a string covered? Same question for C# tests`

Verified we have `IForEachLoopStatement_String` for string case, but we do not have foreach over a collection converted to generic IEnumerable. I will add unit tests to both languages for this scenario.

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #22030

### CODE REVIEW

File: src/Compilers/VisualBasic/Test/IOperation/IOperation/IOperationTests_IForEachLoopStatement.vb [hunk 1]
```diff
@@ -1111,6 +1109,96 @@ IForEachLoopOperation (LoopKind.ForEach, Continue Label Id: 0, Exit Label Id: 1)
             VerifyOperationTreeAndDiagnosticsForTest(Of ForEachBlockSyntax)(source, expectedOperationTree, expectedDiagnostics)
         End Sub
 
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact(), WorkItem(17602, "https://github.com/dotnet/roslyn/issues/17602")>
+        Public Sub IForEachLoopStatement_CastArrayToIEnumerable()
```
Comment (333fred) [inline src/Compilers/VisualBasic/Test/IOperation/IOperation/IOperationTests_IForEachLoopStatement.vb]: No equivalent in C#?

Comment (333fred) [reply src/Compilers/VisualBasic/Test/IOperation/IOperation/IOperationTests_IForEachLoopStatement.vb]: Ah, nevermind, it already existed.

---
In reply to: [198597880](https://github.com/dotnet/roslyn/pull/28158#discussion_r198597880) [](ancestors = 198597880)

**Comment by 333fred** [comment]
Can you verify if we have cfg tests for this scenario?

**Comment by mavasani** [comment]
> Can you verify if we have cfg tests for this scenario?

Yes, we seem to have quite a few of them, such as http://source.roslyn.io/#Roslyn.Compilers.CSharp.Semantic.UnitTests/IOperation/IOperationTests_IForEachLoopStatement.cs,1844 and http://source.roslyn.io/#Roslyn.Compilers.CSharp.Semantic.UnitTests/IOperation/IOperationTests_IForEachLoopStatement.cs,2062. They don't have explicit casts, but do iterate on enumerables, strings, etc. so we should be good.

**Review by 333fred — APPROVED** [review]

**Pull request merged**